### ISSUE TITLE
New Rule: link-fragments-should-be-valid

### ISSUE THREAD

**Issue by SwetaTanwar** [eslint/markdown#369]
### Rule details

This rule should warn when a link fragment does not match any of the fragments that are automatically generated for headings in a document

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
Incorrect Code


# Heading Name

[Link](#fragment)


Correct Code


# Heading Name

[Link](#heading-name)
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Prior At: [Rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md051---link-fragments-should-be-valid)

**Comment by snitin315** [comment]
I'm 👍🏻 for this rule & I feel this should also be recommended.

**Comment by lumirlumir** [comment]
Yes, I agree that this rule is worth implementing.

However, I’m concerned that it has some tricky aspects.

Some Markdown parsers handle heading link fragments in their own ways. For example, one parser might convert `Heading with .dot` to `heading-with-dot`, while another might interpret it as `heading-with-.dot`. This might not be the exact case, but I’ve seen many websites generate link fragments differently.

Additionally, in general usage, we need to consider that many Markdown parsers support custom link fragments like this:

```md
# Heading 1 {#custom-heading-1-it-is-possible-for-many-markdown-parsers}
```

This is commonly used syntax, supported by popular parsers like `markdown-it` and `remark`.

---

So in conclusion, it would be great to implement and recommend this rule if we can account for these cases properly. But until we find a concrete way to handle these situations, I think we should hold off on proceeding and have a discussion about how to implement the rule.

**Comment by SwetaTanwar** [comment]
> I think this is a good idea. @SwetaTanwar Are you planning to submit a PR for this? If not, I can take it on.

Yes, I will be picking it up
